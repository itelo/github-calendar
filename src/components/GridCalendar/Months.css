.month-font {
  font-size: 9px;
  line-height: 1px;
  margin: 0;
  color: #586069;
}

.month-container {
  position: absolute;
  top: -12px;
}
