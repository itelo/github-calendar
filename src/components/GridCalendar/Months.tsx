import * as React from "react";
import "./Months.css";

export type MonthData = {
  month: string;
  position: number;
};

export type MonthsProps = {
  months: MonthData[];
};

export const offsetX = 10;

const Months = (props: MonthsProps) => (
  <React.Fragment>
    {props.months.map(({ month, position }, index) => (
      <div
        key={`month-${index}`}
        data-testid={`month-${index}`}
        className="month-container"
        style={{ left: offsetX * position }}
      >
        <p className="month-font">{month}</p>
      </div>
    ))}
  </React.Fragment>
);

export default Months;
