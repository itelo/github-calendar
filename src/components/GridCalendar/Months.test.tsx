import React from "react";
import { render, fireEvent } from "react-testing-library";
import "react-testing-library/cleanup-after-each";
import Months, { offsetX } from "./Months";
import "jest-dom/extend-expect";

describe("<Months />", () => {
  test("test months passing mocked months", () => {
    const mockMonths = [
      {
        month: "Jan",
        position: 2
      },
      {
        month: "Feb",
        position: 6
      }
    ];
    const { getByTestId, container } = render(<Months months={mockMonths} />);

    mockMonths.forEach((month, index) => {
      const content = getByTestId(`month-${index}`);
      expect(content).toHaveStyle(`left: ${month.position * offsetX}`);
      expect(content).toHaveTextContent(month.month);
    });

    expect(container).toMatchSnapshot();
  });
});
