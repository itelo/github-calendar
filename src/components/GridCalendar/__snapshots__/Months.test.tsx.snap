// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Months /> test months passing mocked months 1`] = `
<div>
  <div
    class="month-container"
    data-testid="month-0"
    style="left: 20px;"
  >
    <p
      class="month-font"
    >
      Jan
    </p>
  </div>
  <div
    class="month-container"
    data-testid="month-1"
    style="left: 60px;"
  >
    <p
      class="month-font"
    >
      Feb
    </p>
  </div>
</div>
`;
